src/Controllers/FeedController.cs

```diff
@@ -32,58 +33,66 @@ public ActionResult RssFeed(string url, int page)
                 var sorterer = new FeedSorterer();
                 sorterer.SortFeed(selectedSort, ref feedItems);
             }
-            int pageSize = int.Parse(_config["FeedPageInfo:pageSize"]);
-            IEnumerable<SyndicationItem> itemsPerPages = feedItems.Skip((page - 1) * pageSize).Take(pageSize);
-            PageInfo pageInfo = new PageInfo { PageNumber = page, PageSize = pageSize, TotalItems = feedItems.Count };
-            Feed rssFeed = _context.Feeds.FirstOrDefault(feed => feed.Url == url);
+            var rssFeed = FormFeed(url, page, feedItems);
+            return View(rssFeed);
+        }
+
+        public Feed FormFeed(string url, int page, List<SyndicationItem> items)
+        {
+            int pageSize = int.Parse(_config.GetSection("FeedPageInfo:pageSize").Value);
+            IEnumerable<SyndicationItem> itemsPerPages = items.Skip((page - 1) * pageSize).Take(pageSize);
+            PageInfo pageInfo = new PageInfo { PageNumber = page, PageSize = pageSize, TotalItems = items.Count };
+            Feed rssFeed = _repository.GetFeedByUrl(url);
             rssFeed.PageInfo = pageInfo;
             rssFeed.SyndicationItems = itemsPerPages;
-            return View(rssFeed);
+            return rssFeed;
         }
 
-        private List<SyndicationItem> GetSyndicationItems(string url)
+        public List<SyndicationItem> GetSyndicationItems(string url)
```

Comment:
So, if methods are public, you need check input arguments
Kinda
if(url.IsNullOrEmpty())
{
   throw new ArgumentNullException(nameog(url));
}


or/and you may check kinda

if(!url.IsValidate())
{
log.Error("Validation failed blablabla")
return error or throw new ValidationExection()
}

do that anywhere for public methods or do methods private